pkg/operator/ceph/cluster/cluster_test.go

```diff
@@ -69,3 +74,71 @@ func TestPreClusterStartValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigureMsgr2(t *testing.T) {
+	type fields struct {
+		encryptionExpected   bool
+		compressionExpected  bool
+		disabledBothExpected bool
+		cephVersion          version.CephVersion
+		Spec                 *cephv1.ClusterSpec
+	}
+	tests := []struct {
+		name   string
+		fields fields
+	}{
+		{"default settings", fields{false, false, false, version.CephVersion{Major: 15}, &cephv1.ClusterSpec{}}},
+		{"encryption enabled", fields{true, false, false, version.CephVersion{Major: 16}, &cephv1.ClusterSpec{Network: cephv1.NetworkSpec{Connections: cephv1.ConnectionsSpec{Encryption: &cephv1.EncryptionSpec{Enabled: true}}}}}},
```

Comment:
Should we test encryption enabled with both old and good version as well?

Reply:
Encryption is available on all supported ceph versions, it's just the compression that isn't supported prior to quincy

Reply:
Oh! Gotcha 👍